frontend/vue/components/CodeExercise/CodeExercise.vue

```diff
@@ -48,6 +49,11 @@ export default defineComponent({
     CodeOutput
   },
   props: {
+    goal: {
+      type: String,
+      required: true,
```

Comment:
should the `goal` property be required? just asking.... could there be scenario where you want grading but dont need progress 🤷🏽

Reply:
Good question, I don't know. I just changed it to not required. I think it is a better implementation. If we want to make sure it exists, we should warn or throw an error in compilation/conversion.